test/prom-query-tests/UnitTest1.cs

```diff
@@ -1,10 +1,132 @@
+    Assert.NotNull(latestresult.Data); // Tests if result data is present
+    var latestresultData = Assert.Single(latestresult.Data.Result);
+    Assert.True(latestresultData.Labels.TryGetValue("__name__", out var sample1labelValue));
+    Assert.Equal(metricName, sample1labelValue); // Tests if metric name and sample 1 label value are the same
+    Assert.True(latestresultData.Value.HasValue); // Tests if result data has a value
+    Assert.True(Double.TryParse(latestresultData.Value.Value.Value, out var sample1MetricValue));
+    Assert.Equal(sample1Value, sample1MetricValue); // Tests if our randomly generated sample 1 Value equals Prom's value
+
+    // Q2: before first/earliest sample (no result)
+    var earliestDataQuery = new QueryPostRequest($"{metricName}", sample2Timestamp.AddMilliseconds(-1), null);
+    var earliestQueryResult = await queryClient.QueryPostAsync(earliestDataQuery, token);
+    Assert.NotNull(earliestQueryResult.Data); // Tests if the oldestquery result data is present
+    Assert.Empty(earliestQueryResult.Data.Result); // Tests if the oldestquery result is empty
+
+    // Q3: before last sample, after first/earliest sample (returns second to last item)
+    var inBetweenQuery = new QueryPostRequest($"{metricName}", lastSampleTime.AddMilliseconds(-1), null);
+    var inBetweenQueryResult = await queryClient.QueryPostAsync(inBetweenQuery, token);
+    Assert.NotNull(inBetweenQueryResult.Data); // Tests if result is present
+    var inBetweenQueryresultData = Assert.Single(inBetweenQueryResult.Data.Result);
+
+    Assert.True(inBetweenQueryresultData.Labels.TryGetValue("__name__", out var sample3labelValue));
+    Assert.Equal(metricName, sample3labelValue); // Tests if both names are equal
+    Assert.True(inBetweenQueryresultData.Value.HasValue); // Tests if result data has a value
+    Assert.True(Double.TryParse(inBetweenQueryresultData.Value.Value.Value, out var sample3MetricValue));
+    Assert.Equal(sample3Value, sample3MetricValue); // Tests if our randomly generated sample 3 Value equals Prom's value
+  }
+
+  private async Task<String> PushTestData(Sample[] samples, CancellationToken token) {
+    // initial setup
+    using var httpClient = CreateHttpClient();
+    var rwClient = new PrometheusRemoteProtocolClient(httpClient);
+
+    // label and value
+    var metricName = $"newval{Guid.NewGuid().ToString().Replace("-", "")}";
+
+    var writeRequest = new WriteRequest();
+    var timeSeries = new TimeSeries();
+    writeRequest.Timeseries.Add(timeSeries);
+    timeSeries.Labels.Add(new Label { Name = "__name__", Value = metricName });
+    var samplesColl = new Google.Protobuf.Collections.RepeatedField<Sample>();
+    timeSeries.Samples.AddRange(samples);
+
+    // push test data to prometheus
+    await rwClient.WriteAsync(writeRequest, token);
+    return metricName;
+  }
+
+  private Sample[] CreateTestData(DateTime seedTime, int numSamples) {
```

Comment:
We should probably create a comment for this function explaining the parameters. Someone who isn't familiar with this code might not understand what `seedTime` and `numSamples` are used for.